src/lib.rs

```diff
@@ -451,7 +423,7 @@ trait HdfsErrorConvert<T> {
     fn to_object_store_err(self) -> Result<T>;
 }
 
-impl<T> HdfsErrorConvert<T> for hdfs_native::Result<T> {
+impl<T> HdfsErrorConvert<T> for client::Result<T> {
```

Comment:
this only throws AlreadyExists error or a Generic error.  
            May be we can also handle InvalidPath, PermissionDenied errors here